source/agora/node/Runner.d

```diff
@@ -143,25 +143,33 @@ public Listeners runNode (Config config)
             //dnstask.interrupt();
         }
 
-    if (config.registry.enabled)
-    {
-        auto reg = result.node.getRegistry();
-        assert(reg !is null);
-        if (hasHTTPInterface)
-            router.registerRestInterface(reg, settings);
-        /* auto dnstask = */ runTask(() => runDNSServer(config.registry, reg));
-        result.tcp ~= listenTCP(config.registry.port, (conn) => conn.runTCPDNSServer(reg),
-                config.registry.address);
-    }
-
     bool delegate (in NetworkAddress address) @safe nothrow isBannedDg = (in address) @safe nothrow {
         try
             return result.node.getBanManager().isBanned(Address("agora://" ~ address.toAddressString()));
         catch (Exception e)
             assert(false, e.msg);
     };
 
-    setTimer(0.seconds, &result.node.start, Periodic.No);  // asynchronous
+    void nodeStarter () @trusted nothrow
+    {
+        try
+        {
+            result.node.start();
+            if (config.registry.enabled)
+            {
+                auto reg = result.node.getRegistry();
+                assert(reg !is null);
+                if (hasHTTPInterface)
+                    router.registerRestInterface(reg, settings);
+                /* auto dnstask = */ runTask(() => runDNSServer(config.registry, reg));
+                result.tcp ~= listenTCP(config.registry.port, (conn) => conn.runTCPDNSServer(reg),
+                        config.registry.address);
+            }
+        } catch (Exception e)
+            assert(false, e.msg);
+    }
+
+    setTimer(0.seconds, &nodeStarter, Periodic.No);  // asynchronous
```

Comment:
`runNode` already runs in a task: https://github.com/bosagora/agora/blob/676ad915df7947607fdf117ba15749a13ff89348/source/agora/node/main.d#L169-L173

Reply:
Then this will be a separate commit.